doc/api/esm.md

```diff
@@ -391,6 +391,32 @@ import { readFileSync } from 'node:fs';
 const buffer = readFileSync(new URL('./data.proto', import.meta.url));
 ```
 
+### `import.meta.main`
+
+<!-- YAML
+added:
+  - REPLACEME
+-->
+
+> Stability: 1.0 - Early development
+
+* {boolean} `true` when the current module is the entry point of the current process; `false` otherwise.
+* Equivalent to `require.main === module` in CommonJS.
+* Analogous to Python's `__name__ == "__main__"`.
+
+```js
+export function foo() {
+  console.log('Hello, world!');
+}
+
+function main() {
+  foo();
+}
+
+if (import.meta.main) main();
+// `foo` can be imported from another module without possible side-effects from `main`
```

Comment:
I don't know if it makes sense to have a `main` that only calls `foo`
```suggestion
if (import.meta.main) {
  console.log('Hello from main');
  foo();
}
```
Or `main` should contain some side-effect so it's more realistic.

Reply:
I was keeping the example minimal. The point was to show `main` is related to `foo` in some sense (since they would not be placed in the same file, otherwise). 

I would prefer not to remove the `main()` function, as it follows Python's pattern and I would like to encourage users to define a `main()` function instead of wrapping their `main()` code in the `if`.

Reply:
Resolved due to https://github.com/nodejs/node/pull/57804/commits/5f18ca6f4a0461f66ff1c868de94db72fd6f4845